onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -123,94 +123,104 @@ Status GraphPartitioner::Partition(Graph& graph, bool export_dll, FuncManager& f
   if (graph.NumberOfNodes() == 0)
     return Status::OK();
 
-  // recurse into nested graphs first so we partition bottom up.
-  for (auto& node : graph.Nodes()) {
-    for (auto& entry : node.GetAttributeNameToMutableSubgraphMap()) {
-      Graph* subgraph = entry.second;
-      // we pass through the export_dll value and FuncManager from the top level graph
-      ORT_RETURN_IF_ERROR(Partition(*subgraph, export_dll, func_mgr));
-    }
-  }
-
-  // fused_kernel_registry is preparing the kernels created on the fly for fused sub graph.
-  // It is only visible for current session.
-  std::shared_ptr<KernelRegistry> fused_kernel_registry = std::make_shared<KernelRegistry>();
-  // Partitioning <graph> based on provider preference and their capabilities.
-  GraphViewer graph_viewer(graph);
-
-  // If an execution provider return the capability that he could run a sub-graph,
-  // onnxruntime will fuse the sub-graph into a function node. if the execution provider
-  // says he need to compile the graph at runtime (by need_compile flag),
-  // onnxruntime will invoke the "Compile" method to get compiled binary.
-  // There are two mode of compile, one is return the entry point to the compiled binary
-  // directly, another is export the compiled binary to shared library for future reuse.
-
-  // TODO: when the graph contain a function node, and user pass in the dll which could
-  // run the function by SessionOption, we should create a function kernel for it and
-  // delegate the compute to the functions inside the dlls.
-  for (auto& provider : providers_) {
-    int count = 0;
-    std::vector<Node*> nodes_need_compile;
-    std::vector<std::unique_ptr<ComputeCapability>> capabilities =
-        provider->GetCapability(graph_viewer, kernel_registry_mgr_.GetKernelRegistriesByProviderType(provider->Type()));
-    for (auto& capability : capabilities) {
-      Node* n = PlaceNode(graph, std::move(capability->sub_graph), kernel_registry_mgr_, provider->Type(), count);
-      if (n != nullptr) {
-        nodes_need_compile.push_back(n);
+  // some feature, for example, function inlining, requires to re-run the partition.
+  // if this flag is set, we need to re-run all the steps until no more re-run is required
+  bool rerun_flag;
+  do {
+    rerun_flag = false;
+
+    // recurse into nested graphs first so we partition bottom up.
```

Comment:
There is NO code change from Line.132 to Line.213 except code-formatting (adding 2 spaces in front of each line)